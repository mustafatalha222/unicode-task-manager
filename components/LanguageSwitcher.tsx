'use client'
import { usePathname, useRouter } from '@/i18n/routing'
import { useLocale, useTranslations } from 'next-intl'
import { Switch } from './ui/switch'

const LanguageSwitcher = () => {
  const t = useTranslations('Lang')
  const router = useRouter()
  const pathname = usePathname()
  const currentLocale = useLocale()

  const changeLanguage = () => {
    const locale = currentLocale === 'en' ? 'ar' : 'en'

    router.replace(pathname, { locale })
  }

  return (
    <div className="flex items-center space-x-2">
      <span className={`font-bold text-sm ${currentLocale === 'ar' ? 'text-primary' : 'text-gray-600'}`}>
        {currentLocale === 'en' ? t('English') : t('Arabic')}
      </span>
      <Switch checked={currentLocale === 'ar'} onCheckedChange={changeLanguage} className="mr-2" />
    </div>
  )
}

export default LanguageSwitcher
