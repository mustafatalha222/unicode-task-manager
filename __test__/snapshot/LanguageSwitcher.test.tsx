import LanguageSwitcher from '@/components/LanguageSwitcher'
import renderer from 'react-test-renderer'
import { NextIntlClientProvider } from 'next-intl'

// Mock translations
const messages = {
  Lang: {
    switchLanguage: 'Switch Language',
  },
}

jest.mock('../../i18n/routing', () => ({
  useRouter() {
    return {
      push: () => jest.fn(),
      replace: () => jest.fn(),
    }
  },
  usePathname() {
    return ''
  },
}))

it('renderer language Switcher unchange', () => {
  const tree = renderer
    .create(
      <NextIntlClientProvider locale="en" messages={messages}>
        <LanguageSwitcher />
      </NextIntlClientProvider>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
