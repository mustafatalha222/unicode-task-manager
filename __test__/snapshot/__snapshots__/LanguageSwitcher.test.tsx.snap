// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer language Switcher unchange 1`] = `
<div
  className="flex items-center space-x-2"
>
  <span
    className="font-bold text-sm text-gray-600"
  >
    Lang.English
  </span>
  <button
    aria-checked={false}
    className="peer inline-flex h-5 w-9 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent shadow-sm transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input mr-2"
    data-state="unchecked"
    onClick={[Function]}
    role="switch"
    type="button"
    value="on"
  >
    <span
      className="pointer-events-none block h-4 w-4 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-4 data-[state=unchecked]:translate-x-0"
      data-state="unchecked"
    />
  </button>
  <input
    aria-hidden={true}
    defaultChecked={false}
    style={
      {
        "margin": 0,
        "opacity": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "transform": "translateX(-100%)",
      }
    }
    tabIndex={-1}
    type="checkbox"
    value="on"
  />
</div>
`;
